src/plugins/intel_cpu/src/nodes/eltwise.cpp

```diff
@@ -1733,8 +1771,67 @@ class EltwiseRefExecutor : public Eltwise::IEltwiseExecutor {
                         *dst_ptr_f = (_opData.alpha && (src_f[0] == -std::numeric_limits<T>::infinity())) ||
                                      (_opData.beta  && (src_f[0] == std::numeric_limits<T>::infinity()));
                         break;
-                    case Algorithm::EltwiseIsNaN:             *dst_ptr_f = std::isnan(src_f[0]); break;
+                    case Algorithm::EltwiseIsNaN: {
+                        if (sizeof(T) == 4) {
+                            *dst_ptr_f = std::isnan(static_cast<float>(src_f[0]));
+                        } else {
+                            *dst_ptr_f = std::isnan(static_cast<dnnl::impl::float16_t>(src_f[0]));
+                        }
+                        break;
+                    }
                     case Algorithm::EltwiseSelect:            *dst_ptr_f = src_f[0] ? src_f[1] : src_f[2]; break;
+                    case Algorithm::EltwiseBitwiseAnd: {
+                        const auto size = sizeof(T);
+                        if (size == 1) {
+                            *bitwise_ptr_cast<uint8_t>(dst_ptr_f) = bitwise_cast<uint8_t>(&src_f[0]) & bitwise_cast<uint8_t>(&src_f[1]);
```

Comment:
Honestly at this point this handling seems too much.
Even we do have long term plans for refactoring eltwise node, I think it's worth to try to address this in scope of this PR. Can we restructure eltwise executor (maybe even introduce Bitwise executor?) to avoid the casting mess.

Reply:
done, refactored